cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java

```diff
@@ -205,14 +205,32 @@ private boolean modelContainsRoot(Model model, Resource subject) {
   private void processContext(Context context, Model model) {
     if (!configuration.isContextEnabled()) return;
 
-    Map<String, String> prefixes = conversionProfile.getContext().getPrefixes();
+    Map<String, String> prefixes = mergePrefixMaps(
+            conversionProfile.getContext().getPrefixes(),
+            model.getNsPrefixMap()
+    );
     if (MapUtils.isEmpty(prefixes)) return;
 
     ObjectNode contextNode = context.jsonRoot.putObject("context");
     ObjectNode prefixNode = contextNode.putObject("prefix");
     prefixes.forEach(prefixNode::put);
   }
 
+  private Map<String, String> mergePrefixMaps(Map<String, String> map1, Map<String, String> map2) {
+    Stream<Map.Entry<String, String>> map2FilteredStream = map2.entrySet().stream()
+                                                    .filter(e -> !map1.containsValue(e.getValue()))
+                                                    .map(e -> {
+                                                      if (!map1.containsKey(e.getKey())) return e;
```

Comment:
maybe a separate method? like newKeyEntry(...)

Reply:
I'll do that, I wasn't sure if it was needed to make a function for something this small. It does look kind of ugly now :/